superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -162,8 +163,6 @@ function dbReducer(
       };
     case ActionType.fetched:
       return {
-        engine: trimmedState.engine,
-        configuration_method: trimmedState.configuration_method,
```

Comment:
why are we removing these?

Reply:
when trying to edit a database, the trimmed state was returning an empty object.

Reply:
ok, then you can move the whole case statement down with 
```
case ActionType.dbSelected:
case ActionType.configMethodChange:
 ```
 since the return value is the same.